### PR TITLE
Update AppVeyor build OS to VS 2017

### LINKED ISSUE #7683: Update AppVeyor build to VS2017?
We are currently testing on ``Visual Studios 2015 update 2`` as indicated [here](https://github.com/astropy/astropy/blob/master/appveyor.yml#L26).
```yml
...
os: Visual Studio 2015 Update 2
...
```
This seems to require an explicit request to use this OS, i.e. it is not in the canonical build list, _c.f._ https://github.com/appveyor/ci/issues/1197

Just prompting the question of whether it's time to update?

### PR DESCRIPTION
**Pull request by jamienoss**
Resolves #7683

Signed-off-by: James Noss <jnoss@stsci.edu>

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -136,6 +136,10 @@ astropy.tests
 - Added an option ``--readonly`` to the test command to change the
   permissions on the temporary installation location to read-only. [#7598]
 
+- Update AppVeyor build OS from ``Visual Studio 2015 Update 2`` to
```
Comment (bsipocz) [inline CHANGES.rst]: no need for changelog here.

Comment (jamienoss) [reply CHANGES.rst]: I'll let the tests run and then change.

**Comment by jamienoss** [comment]
Looks like something broke master
https://travis-ci.org/astropy/astropy/jobs/408189951#L1725

I also experienced this in #7293 https://travis-ci.org/astropy/astropy/jobs/407796141

**Comment by bsipocz** [comment]
Thanks @jamienoss!

**Comment by bsipocz** [comment]
So this started to bite us on LTS. I'm trying to backport and hope that VS2017 works with the python 2.7 job, too.

**Comment by bsipocz** [comment]
well, compiler issues 😞

**Comment by pllim** [comment]
I forgot where I solved a similar problem but instead of updating the version, I removed that line altogether and let Appveyor choose. It was one of the Astropy project repos but I don't remember which one now!

**Comment by bsipocz** [comment]
Ohh, good call. We added it originally for the old image as there were some issues with the current one. I'll try that solution.

**Comment by pllim** [comment]
🤞

**Comment by bsipocz** [comment]
The backport fixed the python3 build, and the other was already set to allowed to fail, so I take this as a success.

**Pull request merged**